File: AFBack/DTOs/MessageRequestCreatedDTO.cs
using System.Text.Json.Serialization;
using AFBack.Models;

namespace AFBack.DTOs;

public class MessageRequestCreatedDto
{
    [JsonPropertyName("senderId")]
    public int SenderId { get; set; }

    [JsonPropertyName("receiverId")]
    public int ReceiverId { get; set; }

    [JsonPropertyName("conversationId")]
    public int ConversationId { get; set; }
    
    // Valgfri notification
    public MessageNotificationDTO? Notification { get; set; }
}

File: shared/types/MessageRequestCreatedDto.ts
// Brukes når en bruker lager en meldingsforespørsel til en annen bruker og sender via backend
import { MessageNotificationDTO } from "./MessageNotificationDTO";
export interface MessageRequestCreatedDto {
  senderId: string;
  receiverId: string;
  conversationId: number;
  notification?: MessageNotificationDTO; // 👈 legg til denne linjen
}